chill-scala/src/main/scala/com/twitter/chill/TupleSerializers.scala

```diff
@@ -699,177 +720,36 @@ class Tuple22Serializer[A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S,
       kser.readClassAndObject(in).asInstanceOf[S],
       kser.readClassAndObject(in).asInstanceOf[T],
       kser.readClassAndObject(in).asInstanceOf[U],
-      kser.readClassAndObject(in).asInstanceOf[V])
+      kser.readClassAndObject(in).asInstanceOf[V]
+    )
   }
 }
-
-class Tuple1LongSerializer extends KSerializer[Tuple1$mcJ$sp] with Serializable {
```

Comment:
wait, removing these will be a performance regression I think (and break people who took their chances and stored these to disk. This set of classes with one type tag tell you both of the tuple types, which will save you from having to write them.

Reply:
Yeah seems these specialized classes were renamed in 2.12.9. I reverted them back.
However those using chill with scala >= 2.12.9 will still experience the regression. Maybe we should remove them in 0.10 once we cross build 2.13?

Reply:
https://github.com/smarter/chill/commits/scala213 handles the tuple serializer thing by rewriting it into Java: https://github.com/smarter/chill/commit/dd1fcd148ba2b8320d90f599cc9d204af13760f7. This should work just as well for 2.12.10, but maintaining binary compat would require some extra effort (which might not be useful in practice, I don't know if references to the tuple serializers are likely to appear in user code)

Reply:
@smarter I assume the classes themselves are still there, you can't can't directly reference them? Otherwise that would be a binary incompatible change in scala 2.12 itself.

I think breaking compatibility would be bad, since it is hard to reason about how these classpaths get set up. Storm, Scalding and Spark all depend on chill. It isn't obvious if they wind up depending on these serializers. Ideally we could make Java types with the same names that would be binary compatible. Maybe that isn't possible.

Reply:
> Ideally we could make Java types with the same names that would be binary compatible. Maybe that isn't possible.

I think it's possible but will require some effort, it might be simpler to keep the current tuple serializers for 2.12-.